### ISSUE TITLE
--dotted is broken

### ISSUE THREAD

**Issue by nikostr** [arvestad/alv#14]
Hey! `--dotted` compares all blocks to the first block of `template_seq`. To fix this, I think all that is needed is updating 

https://github.com/arvestad/alv/blob/88ba4a8b4c1c66fc3d3c5ac914a4edd51cdbcbc2/alv/alignment.py#L159

and

https://github.com/arvestad/alv/blob/88ba4a8b4c1c66fc3d3c5ac914a4edd51cdbcbc2/alv/alignment.py#L285

to

```
if c == template_seq[block.start + col_no]: 
```
and
```
 if c == template_seq[block.start + pos:block.start + pos+3]: 
```
respectively. I've verified the former locally, I'm assuming that the same principle should apply for the latter.

**Comment by nikostr** [comment]
One improvement would be having different background colors for dots and gaps. Currently it is hard to differentiate between them.

**Comment by arvestad** [comment]
Thanks for fixes and improvements! I think the fix in `BaseAlignment` is correct, but taken into `CodonAlignment` actually breaks the code. I had to look this over: in `CodonAlignment`, `template_record` is sliced to the block, but I had missed that in `BaseAlignment`. Or am I missing something?

**Comment by nikostr** [comment]
It does seem to be working as is for the `CodonAlignment`!

Another issue there however is that the background color changes for the dotted residues - if a block starts with a dot, the terminal background color will be used until some non-dot residue turns up, after which the dots will have a white background instead. 

Additional issues:

https://github.com/arvestad/alv/blob/dc07aa49cf8e58ca84d671e140f3c24aa201e7cb/alv/alignment.py#L286

This does not account for length of c varying - replace `'...'` with `'.' * len(c)`. Changing this on its own would color these final dots in the same way as incomplete codons, so perhaps it would make sense to updated the conditionals to account for dots here, to solve both issues? Perhaps like this: 
```
if ('.' not in c) and len(c) != 3:
```
and
```{py}
elif (c == '---') or ('.' in c):
```

 Also, in the code for coloring the incomplete codons, `self.bg_neutral` has been misspelled as `self_bg_neutral` in the pip package.

**Comment by arvestad** [comment]
Yes, I noticed the colouring issue. I will look into your suggestions above as well. Thanks a lot! And it is nice to hear from you again, btw! :-)

**Comment by nikostr** [comment]
Toppen! Ja, verkligen trevligt! Hoppas allt är bra med dig! :)

**Comment by arvestad** [comment]
Brokeness fixed in commit d21211e

**Comment by nikostr** [comment]
Hey! Hope all is well with you :) This seems to have snuck back into v1.5.1 on PyPI.
